packages/ses-ava/src/command.js

```diff
@@ -0,0 +1,155 @@
+/* global process */
+/* eslint-disable no-await-in-loop, no-continue, no-labels, no-unreachable-loop */
+
+/* The ses-ava command allows a single package to run the same test suite with
+ * multiple configurations named in package.json under `sesAvaConfigs`.
+ * Each of these configurations requires a separate Ava config file like
+ * test/_ava.special.config.js and a name.
+ * The ses-ava command will by default run all tests in every mode but allows
+ * the user to pass --only-* and --no-* at any argument position for any of
+ * the named configurations to filter.
+ * Consequently, the "test" script for a package using ses-ava can simply be
+ * "ses-ava" and preserves the filtering behaviors of the underlying "ava"
+ * sessions if run like `yarn test -m file`.
+ */
+
+import popen from 'node:child_process';
+import fs from 'node:fs';
+
+const singlePassThrough = new Set([
+  '-t',
+  '--tap',
+  '-s',
+  '--serial',
+  '-u',
+  '--update-snapshots',
+  '--no-worker-threads',
+  '--verbose',
+  '--no-verbose',
+]);
+
+const doublePassThrough = new Set(['-m', '--match', '--node-arguments']);
+
+export const main = async () => {
+  // Parse configuration.
+  const descriptorText = await fs.promises.readFile('package.json', 'utf8');
+  const descriptor = JSON.parse(descriptorText);
+  const { ava = undefined, sesAvaConfigs = {} } = descriptor;
+  if (ava) sesAvaConfigs.default = undefined;
+  const keys = Object.keys(sesAvaConfigs);
+  const allConfigNames = new Set(keys);
+  const noFlags = new Map(keys.map(key => [`--no-config-${key}`, key]));
+  const onlyFlags = new Map(keys.map(key => [`--only-config-${key}`, key]));
+
+  // Parse arguments.
+  const passThroughArgs = [];
+  const noConfigNames = new Set();
+  const onlyConfigNames = new Set();
+  let failFast = false;
+  let debug = false;
+  let firstArg = true;
+  const argsIterator = process.argv.slice(2)[Symbol.iterator]();
+  for (const arg of argsIterator) {
+    if (arg === '--') {
+      passThroughArgs.push(...argsIterator);
+      break;
+    }
+    const noKey = noFlags.get(arg);
+    const onlyKey = onlyFlags.get(arg);
+    if (singlePassThrough.has(arg)) {
+      passThroughArgs.push(arg);
+    } else if (doublePassThrough.has(arg)) {
+      foundNextArg: {
+        for (const nextArg of argsIterator) {
+          passThroughArgs.push(arg, nextArg);
+          break foundNextArg;
+        }
+        throw new Error(`Expected argument after ${arg}`);
+      }
```

Comment:
This seems like overkill in comparison to just calling `next()`.
```suggestion
      const { done, value } = argsIterator.next();
      if (done) throw new Error(`Expected argument after ${arg}`);
      passThroughArgs.push(arg, value);
```